### ISSUE TITLE
Multi-Host FIX Engine and Library

### ISSUE THREAD

**Issue by markkoszykowski** [real-logic/artio#495]
Working on implementing ```artio``` into a project in a multi-host setup with the FIX Engine (running as a gateway) on one machine and the FIX Library (with app-specific logic) on another. This [Wiki page](https://github.com/real-logic/artio/wiki/Overview#aeron) explains that this should be possible using a UDP Aeron channel, and given ```artio```'s architecture using Aeron (with the option to use IPC or UDP) it makes sense to be configurable in a Single or Multi Host setup.

However, in practice I am having difficulty setting this up. In one Docker container I have configured a basic program running an [```ArchivingMediaDriver```](https://github.com/real-logic/aeron/blob/master/aeron-archive/src/main/java/io/aeron/archive/ArchivingMediaDriver.java) along with a [```FixEngine```](https://github.com/real-logic/artio/blob/master/artio-core/src/main/java/uk/co/real_logic/artio/engine/FixEngine.java). In another container I have configured a [```MediaDriver```](https://github.com/real-logic/aeron/blob/master/aeron-driver/src/main/java/io/aeron/driver/MediaDriver.java) and a [```FixLibrary```](https://github.com/real-logic/artio/blob/master/artio-core/src/main/java/uk/co/real_logic/artio/library/FixLibrary.java) that is polling. I have tried configuring the [```FixEngine.libraryAeronChannel```](https://github.com/real-logic/artio/blob/b6241f335d28ac157492287f4f212d7661ebef78/artio-core/src/main/java/uk/co/real_logic/artio/engine/EngineConfiguration.java#L286) and the [```FixLibrary.libraryAeronChannels```](https://github.com/real-logic/artio/blob/b6241f335d28ac157492287f4f212d7661ebef78/artio-core/src/main/java/uk/co/real_logic/artio/library/LibraryConfiguration.java#L78C26-L78C46) in every unicast configuration using the two hostnames of the containers (see below for details) to no avail.

Looking through the source code, it looks like the [```FixEngine```](https://github.com/real-logic/artio/blob/master/artio-core/src/main/java/uk/co/real_logic/artio/engine/FixEngine.java) and the [```FixLibrary```](https://github.com/real-logic/artio/blob/master/artio-core/src/main/java/uk/co/real_logic/artio/library/FixLibrary.java) were designed to run on the same machine as each component Subscribes and Publishes to this Aeron channel. Perhaps I am misconfiguring or misunderstanding, some direction or perhaps even samples would be much appreciated.

\
\
\
tldr;

Aeron channel configs tried:
| [```FixEngine.libraryAeronChannel```](https://github.com/real-logic/artio/blob/b6241f335d28ac157492287f4f212d7661ebef78/artio-core/src/main/java/uk/co/real_logic/artio/engine/EngineConfiguration.java#L286) | [```FixLibrary.libraryAeronChannels```](https://github.com/real-logic/artio/blob/b6241f335d28ac157492287f4f212d7661ebef78/artio-core/src/main/java/uk/co/real_logic/artio/library/LibraryConfiguration.java#L78C26-L78C46) | Issue |
|:------------:|:------------:|:------------|
| ```aeron:udp?endpoint=<engine_host>:<port>``` | [ ```aeron:udp?endpoint=<engine_host>:<port>``` ] | <ul><li>Library fails with a ```java.net.BindException```</li></ul> |
| ```aeron:udp?endpoint=<engine_host>:<port>``` | [ ```aeron:udp?endpoint=<library_host>:<port>``` ] | <ul><li>Engine and Library never connect</li></ul> |
| ```aeron:udp?endpoint=<library_host>:<port>``` | [ ```aeron:udp?endpoint=<engine_host>:<port>``` ] | <ul><li>Library fails with a ```java.net.BindException```</li><li> Engine never initializes and halts</li></ul> |
| ```aeron:udp?endpoint=<library_host>:<port>``` | [ ```aeron:udp?endpoint=<library_host>:<port>``` ] | <ul><li>Engine never initializes and halts</li></ul> |

<sub>(I am aware that setting different Aeron channels would not even make sense, however I have tested and listed them to be as thorough as possible.)</sub>
